### ISSUE TITLE
[bitnami/rabbitmq] metrics.podAnnotations can not be changed

### ISSUE THREAD

**Issue by bmailhe** [bitnami/charts#6026]
**Which chart**:
[bitnami/rabbitmq] 8.11.5 

**Describe the bug**
It is not possible to remove `metrics` pod annotations in statufulset. We want to remove prometheus annotations by setting `metrics.podAnnotations: {}` but does not produce any effect

**To Reproduce**

 - create a values file to enable `metrics` and remove `podAnnotations` :
````
cat values.metrics.yaml
metrics:
  enabled: true
  podAnnotations: {}
````
 - generate the statefulset template : 
````
helm template . -f values.yaml  -s templates/statefulset.yaml -f values.metrics.yaml| yq e ".spec.template.metadata.annotations" -
checksum/config: b4a5718266b60ba94fccb12fc83233bf2bed22835e50b78f28aca5b76bcea5c7
checksum/secret: 33ba567147651d0d0b13435eefe50e210b47de861736875307c64e33dc26fe2e
prometheus.io/port: '9419'
prometheus.io/scrape: "true"
````

According to the _helpers.tpl, annotations `prometheus*` should not be present:
````
{{/*
Return podAnnotations
*/}}
{{- define "rabbitmq.podAnnotations" -}}
{{- if .Values.podAnnotations }}
{{ include "common.tplvalues.render" (dict "value" .Values.podAnnotations "context" $) }}
{{- end }}
{{- if .Values.metrics.enabled }}
{{ include "common.tplvalues.render" (dict "value" .Values.metrics.podAnnotations "context" $) }}
{{- end }}
{{- end -}}
````
**Expected behavior**
When setting 
````
metrics:
  enabled: true
  podAnnotations: {}
````
the statusfulset should not have any `metrics` pod annotations.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.5.1", GitCommit:"32c22239423b3b4ba6706d450bd044baffdcf9e6", GitTreeState:"clean", GoVersion:"go1.15.7"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.4-dirty", GitCommit:"e87da0bd6e03ec3fea7933c4b5263d151aafd07c", GitTreeState:"dirty", BuildDate:"2021-03-15T09:58:13Z", GoVersion:"go1.16.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.17", GitCommit:"f3abc15296f3a3f54e4ee42e830c61047b13895f", GitTreeState:"clean", BuildDate:"2021-01-13T13:13:00Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by alvneiayu** [comment]
hello @bmailhe 

I am trying to reproduce the issue. Please, could you share with me your values.yaml?

Thanks

Álvaro

**Comment by bmailhe** [comment]
Hello @alvneiayu , 

The `values.yaml` is the one provided by the chart without modification. The way to reproduce is to take the chart out of the box and just add the `values.metrics.yaml`.

Thanks !

**Comment by alvneiayu** [comment]
Hi @bmailhe 

Following the helm documentation:

https://helm.sh/docs/chart_template_guide/values_files/

The best way to override a value defined by default is to use "null" instead of {} that you have used inside the values.metric.yaml

```
metrics:
  enabled: true
  podAnnotations: null
```

Doing this, I found an issue and I already sent a patch to fix the problem. With this change integrated and changing the podAnnotions to null, your problem must be fixed.

I hope that I have helped you and thanks for your time.

Álvaro
